packages/kit/src/exports/vite/index.js

```diff
@@ -603,45 +650,24 @@ function kit({ svelte_config }) {
 					})};\n`
 				);
 
+				// regenerate nodes with the client manifest...
+				build_server_nodes(out, kit, manifest_data, server_manifest, client_manifest, css);
+
+				// ...and prerender
 				log.info('Prerendering');
-				await new Promise((fulfil, reject) => {
-					const results_path = `${kit.outDir}/generated/prerendered.json`;
-
-					// do prerendering in a subprocess so any dangling stuff gets killed upon completion
-					const script = fileURLToPath(new URL('../../core/postbuild/index.js', import.meta.url));
-
-					const child = fork(
-						script,
-						[
-							vite_config.build.outDir,
-							manifest_path,
-							results_path,
-							'' + verbose,
-							JSON.stringify({ ...env.private, ...env.public })
-						],
-						{
-							stdio: 'inherit'
-						}
-					);
 
-					child.on('exit', (code) => {
-						if (code) {
-							reject(new Error(`Prerendering failed with code ${code}`));
-						} else {
-							const results = JSON.parse(fs.readFileSync(results_path, 'utf8'), (key, value) => {
-								if (key === 'pages' || key === 'assets' || key === 'redirects') {
-									return new Map(value);
-								}
-								return value;
-							});
-
-							prerendered = results.prerendered;
-							prerender_map = new Map(results.prerender_map);
-
-							fulfil(undefined);
-						}
-					});
-				});
+				await fork('../../core/postbuild/prerender.js', [
+					out,
+					manifest_path,
+					'' + verbose,
+					JSON.stringify({ ...env.private, ...env.public })
+				]);
+
+				const results = devalue.parse(read(`${out}/prerendered.json`));
```

Comment:
My idea wouldn't be to get rid of the filesystem writes, rather encapsulate that logic. For example this would become `await prerender(...)` and inside `prerender.js` it's something like
```js
const [theargs] = process.argv;
if (theargs.something) {
  // we're invoked by ourselves, do the subprocess stuff
  stuff(..);
}

function stuff(..) { 
  // ..
  // write to prerendered.json
}

export async function prerender(..) {
  await fork('./prerender.js', ...);
  // read the file we just wrote to, delete it and return result
}
```

It's also weird because the file is calling itself but at least the public API is more visible.

I'm not sure to what extent this approach could be used for the other intermediate files, so if that only is possible for `prerendered.json` it's probably not of much use to change it.